File: src/main/java/io/felipeandrade/gsw2/material/vanilla/CoalMaterial.java
package io.felipeandrade.gsw2.material.vanilla;

import net.minecraft.item.Item;

import java.util.Collections;
import java.util.List;

import io.felipeandrade.gsw2.GSWItemGroup;
import io.felipeandrade.gsw2.common.GSWColor;
import io.felipeandrade.gsw2.item.GSWItem;
import io.felipeandrade.gsw2.material.ColorProviderMaterialItem;
import io.felipeandrade.gsw2.material.GSWMaterial;
import io.felipeandrade.gsw2.material.GSWMaterialItem;

public class CoalMaterial extends GSWMaterial {

    public static final GSWMaterial MATERIAL = new CoalMaterial();

    public CoalMaterial() {
        super("coal", GSWColor.jet_grey_gray);
    }

    @Override
    public List<? extends GSWItem> allItems() {
        return Collections.singletonList(DUST);
    }


    public static final GSWMaterialItem DUST = new ColorProviderMaterialItem("dust", MATERIAL, new Item.Settings().group(GSWItemGroup.MATERIALS));

}

File: gsw-mod/src/main/kotlin/io/felipeandrade/gsw/material/vanilla/CoalMaterial.kt
package io.felipeandrade.gsw.material.vanilla

import io.felipeandrade.gsw.item.GSWItem
import io.felipeandrade.gsw.material.GSWMaterial
import io.felipeandrade.gsw.material.GSWMaterialItem
import net.fabricmc.fabric.api.datagen.v1.provider.FabricRecipeProvider
import net.minecraft.data.server.recipe.RecipeExporter
import net.minecraft.item.Item
import net.minecraft.item.ItemGroup

class CoalMaterial : GSWMaterial("coal") {

    override fun allItems(): List<GSWItem> = listOf(DUST, CRUSHED)
    override fun generateRecipes(provider: FabricRecipeProvider, exporter: RecipeExporter) {

    }

    companion object {
        val MATERIAL: GSWMaterial = CoalMaterial()
        val DUST: GSWMaterialItem = GSWMaterialItem("dust", MATERIAL, Item.Settings())
        val CRUSHED: GSWMaterialItem = GSWMaterialItem("crushed", MATERIAL, Item.Settings())

        fun addMaterialsToItemGroup(entries: ItemGroup.Entries): Unit = with(entries) {
            add(DUST)
            add(CRUSHED)
        }

        fun addToolsToItemGroup(entries: ItemGroup.Entries) {}
    }
}